sdk/typescript/src/dwallet-mpc/globals.ts

```diff
@@ -139,3 +139,45 @@ export const getEventByTypeAndSessionId = async (
 		} minutes (${seconds} seconds passed).`,
 	);
 };
+
+export async function fetchCompletedEvent<TEvent extends { session_id: string }>(
+	c: Config,
+	sessionID: string,
+	eventType: string,
+	isEventFn: (parsedJson: unknown) => parsedJson is TEvent,
+): Promise<TEvent> {
+	const startTime = Date.now();
+	let cursor = null;
+
+	while (Date.now() - startTime <= c.timeout) {
+		// Wait for a bit before polling again, objects might not be available immediately.
+		await new Promise((resolve) => setTimeout(resolve, 5_000));
```

Comment:
You can use the timeout from the config

Reply:
No. we wait here five seconds before polling, not ten minutes (the timeout) before start polling.